content/guides/comparators.adoc

```diff
@@ -0,0 +1,635 @@
+= Comparators in Clojure
+Andy Fingerhut
+2013-04-01
+:jbake-type: page
+:toc: macro
+
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+
+== Versions
+
+This document was written while checking the details against Clojure 1.5.1  and Java 6,
+but most or all of it should apply to other versions, too.
+
+== Summary
+
+A comparator is a function that takes two arguments _x_ and _y_ and returns a value indicating
+the relative order in which _x_ and _y_ should be sorted. It can be a 3-way comparator returning
+an integer, or a 2-way comparator returning a boolean.
+See the DOs below for what the return values should be, depending upon the order of _x_ and _y_.
+
+In Clojure you need comparators for sorting a collection of values, or for maintaining a collection
+of values in a desired sorted order, e.g a
+http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/sorted-map[sorted-map],
+http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/sorted-set[sorted-set], or
+http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/priority-map[priority-map]
```

Comment:
There is no clojure.core/priority-map - this should refer to clojure.data.priority-map (in the contrib library data.priority-map).